woocommerce/woocommerce
# Fix the current order billing or shipping country not being available for edit if that country is no longer allowed

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

When an old order billing or shipping country is no longer allowed, if the shop admin enters that order edit screen and hits "Update", the country will be lost, because the country edit selects are only showing the current allowed countries.

This PR created two new WC_Order methods that will get the current allowed countries and add the current order country to the returned array, if not present, so the select field contains the original country. Both for billing and shipping.

Closes #41777 



### How to test the changes in this Pull Request:

1. Allow WooCommerce to sell to countries A, B and C
2. Create an order to country C
3. Visit the order edit screen for that order: all should be OK
4. Go to the WooCommerce General settings and remove country C
5. If this PR is merged, country C will still be there on the billing or shipping country select field. Without this PR, it wouldn't be there.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Fixes missing billing or shipping country in the address select fields on the edit order screen if the country is no longer allowed

</details>
